### PR TITLE
add the missing metrics exporter: exporter-metrics-otlp-http

### PR DESCRIPTION
**Pull request by stevemao**
Otherwise metrics won't work

### CODE REVIEW

File: nodejs/packages/layer/src/wrapper.ts [hunk 1]
```diff
@@ -121,8 +122,12 @@ async function initializeProvider() {
   tracerProvider.register(sdkRegistrationConfig);
 
   // Configure default meter provider (doesn't export metrics)
+  const metricExporter = new OTLPMetricExporter();
   let meterConfig: MeterProviderOptions = {
     resource,
+    readers: [new PeriodicExportingMetricReader({
```
Comment (sashaaKr) [inline nodejs/packages/layer/src/wrapper.ts:128]: @stevemao I think we need to import `PeriodicExportingMetricReader` from '@opentelemetry/sdk-metrics'

Comment (stevemao) [reply nodejs/packages/layer/src/wrapper.ts:128]: fixed

**Comment by stevemao** [comment]
@rapphil please review this PR

**Comment by sashaaKr** [comment]
@stevemao I published a custom layer based on this PR to test it in our lambda and it doesn't work,. The `configureMeter` function if `undefined`, the meter has not been configured.

It is unclear where these functions come from, they are supposed to be defined in a global scope, but neither `opentelemetry-lambda`, `opentelemetry-js` nor `opentelemetry-js-contrib` export those functions.

**Comment by onhate** [comment]
I need this too

File: nodejs/packages/layer/src/wrapper.ts [hunk 2]
```diff
@@ -22,7 +22,8 @@ import {
 import { getEnv } from '@opentelemetry/core';
 import { AwsLambdaInstrumentationConfig } from '@opentelemetry/instrumentation-aws-lambda';
 import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-proto';
-import { MeterProvider, MeterProviderOptions } from '@opentelemetry/sdk-metrics';
+import { MeterProvider, MeterProviderOptions, PeriodicExportingMetricReader } from '@opentelemetry/sdk-metrics';
+import { OTLPMetricExporter } from '@opentelemetry/exporter-metrics-otlp-grpc'
```
Comment (martinkuba) [inline nodejs/packages/layer/src/wrapper.ts:26]: I think this should be exporter from `exporter-metrics-otlp-proto`. That's what what is in package.json, and the tracer is using HTTP proto as well.

File: nodejs/packages/layer/src/wrapper.ts (hunk 1 above)
Comment (martinkuba) [reply nodejs/packages/layer/src/wrapper.ts:128]: I think the reader should be added only if the `configureMeterProvider()` function is not defined.

File: nodejs/packages/layer/src/wrapper.ts (hunk 2 above)
Comment (stevemao) [reply nodejs/packages/layer/src/wrapper.ts:26]: what's the differences between the 2 packages?

**Comment by vinay2897** [comment]
This would be very helpful

File: nodejs/packages/layer/src/wrapper.ts (hunk 2 above)
Comment (martinkuba) [reply nodejs/packages/layer/src/wrapper.ts:26]: Both use protobuf as the data format, but `exporter-*-otlp-proto` uses HTTP as the protocol.

Comment (tylerbenson) [reply nodejs/packages/layer/src/wrapper.ts:26]: @stevemao Do you have any objection to this suggestion?

Comment (stevemao) [reply nodejs/packages/layer/src/wrapper.ts:26]: Can make the changes

**Comment by lasith-kg** [comment]
This is blocking me from using Custom Metrics in Lambda. Would be awesome if this could get more traction! Otherwise, I would like to know why custom metrics have been disabled for the NodeJS Lambda Layer.

**Comment by tylerbenson** [comment]
@lasith-kg We are waiting for the author to implement the suggestion from Martin, and rebase on main.

**Comment by lasith-kg** [comment]
> @lasith-kg We are waiting for the author to implement the suggestion from Martin, and rebase on main.

No stress! I've managed to reach the same effect of a Lambda Layer through Manual Instrumentation in the meanwhile. @stevemao for reference here is what I wrote to achieve this. 

The one thing to note is I had to invoke `metrics.setGlobalMeterProvider(meterProvider)` to insure I could generate metric meters globally. Something that doesn't need to be done for the traces provider for some reason.

```
/*instrumentation.js*/
const { metrics, trace, diag, DiagConsoleLogger, DiagLogLevel } = require('@opentelemetry/api');
const { ExpressInstrumentation } = require('@opentelemetry/instrumentation-express');
const { HttpInstrumentation } = require("@opentelemetry/instrumentation-http");
const { UndiciInstrumentation } = require('@opentelemetry/instrumentation-undici');
const { AwsInstrumentation } = require("@opentelemetry/instrumentation-aws-sdk");
const { AwsLambdaInstrumentation } = require('@opentelemetry/instrumentation-aws-lambda')
const { registerInstrumentations } = require('@opentelemetry/instrumentation');
const { NodeTracerProvider, BatchSpanProcessor } = require('@opentelemetry/sdk-trace-node');
const { OTLPTraceExporter } = require('@opentelemetry/exporter-trace-otlp-proto')
const {
  detectResourcesSync,
  envDetector,
  processDetector,
} = require('@opentelemetry/resources');
const { awsLambdaDetector } = require('@opentelemetry/resource-detector-aws');
const { AWSXRayPropagator } = require('@opentelemetry/propagator-aws-xray');
const { AWSXRayIdGenerator } = require('@opentelemetry/id-generator-aws-xray');
const { MeterProvider, PeriodicExportingMetricReader } = require('@opentelemetry/sdk-metrics');
const { OTLPMetricExporter } = require('@opentelemetry/exporter-metrics-otlp-proto')

// Register instrumentations synchronously to ensure code is patched even before provider is ready.
const instrumentations = [
  new HttpInstrumentation(),
  new ExpressInstrumentation(),
  new UndiciInstrumentation(),
  new AwsInstrumentation({suppressInternalInstrumentation: true}),
  new AwsLambdaInstrumentation(),
]
registerInstrumentations({instrumentations})

const resource = detectResourcesSync({
  detectors: [awsLambdaDetector, envDetector, processDetector],
});

let config = {
  resource,
  idGenerator: new AWSXRayIdGenerator(),
};
let sdkRegistrationConfig = {
  propagator: new AWSXRayPropagator(),
};

const tracerProvider = new NodeTracerProvider(config);
const tracerExporter = new OTLPTraceExporter()
const tracerProcessor = new BatchSpanProcessor(tracerExporter)

tracerProvider.addSpanProcessor(tracerProcessor);
tracerProvider.register(sdkRegistrationConfig);

const metricExporter = new OTLPMetricExporter({});
const metricReader = new PeriodicExportingMetricReader({
  exporter: metricExporter,
})
const meterProvider = new MeterProvider({
  resource,
  readers: [metricReader]
});

metrics.setGlobalMeterProvider(meterProvider)

registerInstrumentations({
  instrumentations,
  tracerProvider,
  meterProvider
})
```

```
/* index.js */
const { metrics, trace } = require('@opentelemetry/api');

const meter = metrics.getMeter(process.env.OTEL_SERVICE_NAME);
const counter = meter.createCounter('requests');

const tracer = trace.getTracer(process.env.OTEL_SERVICE_NAME);
const span  = tracer.startSpan('work')
...
span.end()
```

```
node --require ./instrumentation.js index.js
```

**Comment by stevemao** [comment]
I'm not sure what I need to change. Please submit your own PR to address any issue. Cheers!

File: nodejs/packages/layer/src/wrapper.ts (hunk 2 above)
Comment (tylerbenson) [reply nodejs/packages/layer/src/wrapper.ts:26]: @stevemao below you said
> I'm not sure what I need to change. Please submit your own PR to address any issue. Cheers!

This is the change I was referring to.  Seems like a straightforward suggestion. If you have questions, you can also stop by the SIG meeting to discuss.

**Comment by pragmaticivan** [comment]
This can be closed

**Comment by tylerbenson** [comment]
@pragmaticivan can you link to the PR that superseded this?

**Comment by pragmaticivan** [comment]
@tylerbenson https://github.com/open-telemetry/opentelemetry-lambda/pull/1464/files

**Pull request closed without merging**